pkg/kubectl/apply/list_element.go

```diff
@@ -65,3 +67,17 @@ func sliceCast(i interface{}) []interface{} {
 	}
 	return i.([]interface{})
 }
+
+// DetectConflict returns ConflictError if fields in recorded and remote of ListElement conflict
+// ConflictError Message contains details of conflict.
+func (e ListElement) DetectConflict() error {
+	for _, item := range e.Values {
+		if IsConflict(item) {
+			return &ConflictError{Message: fmt.Sprintf("Conflict detected in list elements, recorded value (%+v) and remote value (%+v)",
```

Comment:
nit, but the message says `list elements` here, rather than `ListElement`.  A later message below uses `MapElement`.
Keeping these messages consistent and code a bit more concise would be a good case for something like this helper:

```go
func newConflictError(e Element) *ConflictError {
	return &ConflictError{
		message: fmt.Sprintf("conflict detected in %s, recorded value (%+v) and remote value (%+v)",
			e.GetFieldType(), e.GetRecorded(), e.GetRemote()),
	}
}
```

Also note `conflict` rather than `Conflict`: https://github.com/golang/go/wiki/CodeReviewComments#error-strings